ims/einstein/iscsi/tgt.py

```diff
@@ -11,180 +12,154 @@
             else:
-                raise iscsi_exceptions.NodeAlreadyInUseException()
+                raise iscsi_exceptions.TargetExistsException()
         except (IOError, OSError) as e:
-            logger.info("Update config exception")
-            raise iscsi_exceptions.UpdateConfigFailedException(str(e))
-        except (iscsi_exceptions.MountException,
-                iscsi_exceptions.DuplicatesException) as e:
-            logger.info("Error exposing iscsi_target")
-            raise e
+            raise iscsi_exceptions.TargetCreationFailed(str(e))
+        except subprocess.CalledProcessError as e:
+            raise iscsi_exceptions.TargetCreationFailed(str(e))
 
     @log
     def remove_target(self, target_name):
-        '''
-        Removes target specified. Same as comment for above function.
-        :return:
-        '''
+        """
+        Removes target specified
+
+        :param target_name: Name of target to be removed
+        :return: None
+        """
         try:
             targets = self.list_targets()
             if target_name in targets:
-                os.remove(self.TGT_ISCSI_CONFIG + target_name + ".conf")
-                # tgtarglist = ["sudo", "-S","tgt-admin", "--execute"]
+                os.remove(os.path.join(self.TGT_ISCSI_CONFIG,
+                                       target_name + ".conf"))
                 command = "tgt-admin -f --delete {0}".format(target_name)
-                proc = subprocess.Popen(command, shell=True,
-                                        stdout=subprocess.PIPE,
-                                        stderr=subprocess.PIPE)
-                proc.communicate()
-                if proc.returncode != 0:
-                    raise iscsi_exceptions.UpdateConfigFailedException(
-                        "Deleting target failed at deleting target file stage")
+                output = shell.call(command, sudo=True)
+                logger.debug("Output = %s", output)
             else:
-                raise iscsi_exceptions.NodeAlreadyUnmappedException()
-                # The above should be something like NodeAlreadyDeleted
+                raise iscsi_exceptions.TargetDoesntExistException()
         except (IOError, OSError) as e:
-            # For checking if we have access to directory for file
-            # creation/deletion
-            logger.info("Update config exception")
-            raise iscsi_exceptions.UpdateConfigFailedException(e.message)
-        except (iscsi_exceptions.MountException,
-                iscsi_exceptions.DuplicatesException) as e:
-            # Do we still need the above exceptions? Duplicate
-            # exception can be avoided by checking lists as shown above
-            # Mouting is something that we are not doing now?
-            logger.info("Error exposing iscsi_target")
-            raise e
+            raise iscsi_exceptions.TargetDeletionFailed(str(e))
+        except subprocess.CalledProcessError as e:
```

Comment:
While I agree that the exception to be caught here shouldn't be ``subprocess.CalledProcessError``, you would still need to raise another exception in ``shell.call()`` and catch that one here.